sycl/include/sycl/detail/pi.h

```diff
@@ -2353,6 +2445,222 @@ piextEnqueueCommandBuffer(pi_ext_command_buffer command_buffer, pi_queue queue,
+
+/// API to create samplers for bindless images.
+///
+/// \param context is the pi_context
+/// \param device is the pi_device
+/// \param sampler_properties is the pointer to the sampler properties bitfield
+/// \param min_mipmap_level_clamp is the minimum mipmap level to sample from
+/// \param max_mipmap_level_clamp is the maximum mipmap level to sample from
+/// \param max_anisotropy is the maximum anisotropic ratio
+/// \param result_sampler is the returned sampler
+__SYCL_EXPORT pi_result piextBindlessImageSamplerCreate(
+    pi_context context, const pi_sampler_properties *sampler_properties,
+    float min_mipmap_level_clamp, float max_mipmap_level_clamp,
+    float max_anisotropy, pi_sampler *result_sampler);
+
+/// API to copy image data Host to Device or Device to Host.
+///
+/// \param queue is the queue to submit to
+/// \param dst_ptr is the location the data will be copied to
+/// \param src_ptr is the data to be copied
+/// \param image_format format of the image (channel order and data type)
+/// \param image_desc image descriptor
+/// \param flags flags describing copy direction (H2D or D2H)
+/// \param src_offset is the offset into the source image/memory
+/// \param dst_offset is the offset into the destination image/memory
+/// \param copy_extent is the extent (region) of the image/memory to copy
+/// \param host_extent is the extent (region) of the memory on the host
+/// \param num_events_in_wait_list is the number of events in the wait list
+/// \param event_wait_list is the list of events to wait on before copying
+/// \param event is the returned event representing this operation
+__SYCL_EXPORT pi_result piextMemImageCopy(
+    pi_queue command_queue, void *dst_ptr, void *src_ptr,
+    const pi_image_format *image_format, const pi_image_desc *image_desc,
+    const pi_image_copy_flags flags, pi_image_offset src_offset,
+    pi_image_offset dst_offset, pi_image_region copy_extent,
+    pi_image_region host_extent, pi_uint32 num_events_in_wait_list,
+    const pi_event *event_wait_list, pi_event *event);
+
+/// API to query an image memory handle for specific properties.
+///
+/// \param mem_handle is the handle to the image memory
+/// \param param_name is the queried info name
+/// \param param_value is the returned query value
+/// \param param_value_size_ret is the returned query value size
+__SYCL_EXPORT pi_result piextMemImageGetInfo(
+    const pi_image_mem_handle mem_handle, pi_image_info param_name,
+    void *param_value, size_t *param_value_size_ret);
+
+/// API to import external memory in the form of a file descriptor.
+///
+/// \param context is the pi_context
+/// \param device is the pi_device
+/// \param size is the size of the external memory
+/// \param file_descriptor is the file descriptor
+/// \param ret_handle is the returned interop memory handle to the external
+/// memory
+__SYCL_EXPORT pi_result
+piextMemImportOpaqueFD(pi_context context, pi_device device, size_t size,
```

Comment:
this API is to import the handle,. What is the API to export it?

Reply:
Currently there is no API to export SYCL memory or event objects, as none of the backends that support our bindless images extension also support the exportation of memory or semaphores/events.

Reply:
@przemektmalon : L0 backend supports already the export of memory handles, including images, so we have some support there. Additionally, that shouldn't be a blocker to define it in SYCL and in the UR. Moreover, having it defined in SYCL and UR would provide a template on how to define it in the other backends.

W/o the export API this SYCL extension seems then incomplete. Basically, we are defining the interfaces for interop, but only the import API, not the export :/